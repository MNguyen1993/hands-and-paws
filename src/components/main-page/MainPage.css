@import url('https://fonts.googleapis.com/css?family=Baloo+Bhai|Pacifico&display=swap');

h1 {
	font-family: 'Pacifico';
	color: #4bcffa;
}

body {
	font-family: 'Baloo Bhai';
}
