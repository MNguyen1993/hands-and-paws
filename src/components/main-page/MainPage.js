import React, { Component } from 'react';
import CardList from '../card-list/CardList';
import SearchBox from '../search-box/SearchBox';
import Scroll from '../scroll/Scroll';
import Header from '../header/Header';
import ErrorBoundary from '../error-boundary/ErrorBoundary';
import './MainPage.css';

class MainPage extends Component {
	componentDidMount() {
		this.props.onRequestPets();
	}

	filterPets = pets => {
		return pets.filter(pet => {
			return pet.name
				.toLowerCase()
				.includes(this.props.searchField.toLowerCase());
		});
	};

	render() {
		const { onSearchChange, pets, isPending } = this.props;

		return (
			<div className="tc">
				<Header />
				<SearchBox searchChange={onSearchChange} />
				<Scroll>
					{isPending ? (
						<h1>Loading</h1>
					) : (
						<ErrorBoundary>
							<CardList pets={this.filterPets(pets)} />
						</ErrorBoundary>
					)}
				</Scroll>
			</div>
		);
	}
}

export default MainPage;
