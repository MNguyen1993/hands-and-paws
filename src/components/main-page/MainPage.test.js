import { shallow } from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

describe('MainPage Component', () => {
	const mockProps = {
		onRequestPets: jest.fn(),
		pets: [
			{
				id: 3,
				name: 'John',
				email: 'john@gmail.com',
			},
		],
		searchField: 'a',
		isPending: false,
	};

	const wrapper = shallow(<MainPage {...mockProps} />);

	it('renders component correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('filters pets correctly', () => {
		expect(wrapper.instance().filterPets([])).toEqual([]);
		expect(wrapper.instance().filterPets(mockProps.pets)).toEqual([]);
	});

	it('Scroll component renders H1 with isPending is true', () => {
		const mockProps2 = {
			onRequestPets: jest.fn(),
			pets: [],
			searchField: '',
			isPending: true,
		};

		const wrapper2 = shallow(<MainPage {...mockProps2} />);

		expect(wrapper2.find('Scroll h1').text()).toEqual('Loading');
	});
});
