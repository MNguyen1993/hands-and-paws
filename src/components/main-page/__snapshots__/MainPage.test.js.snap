// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage Component renders component correctly 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundary>
      <CardList
        pets={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
